pkg/util/fedinformer/genericmanager/single-cluster-manager.go

```diff
@@ -87,8 +87,11 @@ type singleClusterInformerManagerImpl struct {
 }
 
 func (s *singleClusterInformerManagerImpl) ForResource(resource schema.GroupVersionResource, handler cache.ResourceEventHandler) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
```

Comment:
I don't get what benefits we can get by this `mutex` lock.

Reply:
I found it is bug, cc test in #3122